Sources/NIO/ServerSocket.swift

```diff
@@ -109,4 +109,17 @@
             return sock
         }
     }
+    
+    /// Close the socket.
+    ///
+    /// After the socket was closed all other methods will throw an `IOError` when called.
+    ///
+    /// - throws: An `IOError` if the operation failed.
+    override func close() throws {
+        let maybePathname = try? self.localAddress().pathname
```

Comment:
This extra syscall on shutdown is a bit of a pain. Can we add a stored property to `Socket` that indicates whether we think we have to unlink it?

Reply:
we could, but with the current `Socket` only storing `descriptor` adding the full string path representation seemed like a lot to add

Reply:
We don't need to store that, we just need to store a bool. I want to avoid us paying this syscall for _every_ socket in the program, including all ephemeral TCP sockets. I'm fine paying for the syscall for sockets that actually require it.

Reply:
Ah, though I double-checked and this is only on `ServerSocket`. So I don't feel that bad, actually. Let's just leave this.